Library/Homebrew/formula_installer.rb

```diff
@@ -1226,8 +1226,27 @@ def pour
     cellar = formula.bottle_specification.tag_to_cellar(Utils::Bottles.tag)
     return if [:any, :any_skip_relocation].include?(cellar)
 
-    prefix = Pathname(cellar).parent.to_s
-    return if cellar == HOMEBREW_CELLAR.to_s && prefix == HOMEBREW_PREFIX.to_s
+    cellar = Pathname(cellar)
+    prefix = cellar.parent
+    return if cellar.to_s == HOMEBREW_CELLAR.to_s && prefix.to_s == HOMEBREW_PREFIX.to_s
+
+    if cellar.realpath.to_s == HOMEBREW_CELLAR.realpath.to_s && prefix.realpath.to_s == HOMEBREW_PREFIX.realpath.to_s
+      opoo <<~EOS
+        Your cellar path contains symlinks which may cause problems:
+        - HOMEBREW_CELLAR: #{HOMEBREW_CELLAR} (#{cellar})
+        - HOMEBREW_PREFIX: #{HOMEBREW_PREFIX} (#{prefix})
+        Consider installing Homebrew into a real directory.
+      EOS
+      if OS.linux?
+        opoo <<~EOS
+          You can ask an admin to replace the symlinks created by:
+            sudo ln -s /src/folder /target/folder
+          with the `mount` command:
+            sudo mount --bind /src/folder /target/folder
+        EOS
+      end
```

Comment:
I add a hint for replacing the symlinks with a bind mount.

Reply:
In all these cases they require root access (from someone) and that root access could be used to install Homebrew in the default location instead?

Reply:
The Homebrew installer doesn't warn users that they are installing Homebrew into a path with symlinks. The installation would be successful with the default prefix `/home/linuxbrew/.linuxbrew`. However, Homebrew refuses to pour bottles for users due to the symlinks.

Maybe we can move these warnings to Homebrew/install rather than here.

Reply:
> Maybe we can move these warnings to Homebrew/install rather than here.

Yeh, I think that'd be the best place. Even then: I'd definitely want to see multiple users requesting this before we do so. We should also perhaps just consider (again) making the installer not install to non-default prefixes that we don't support.